### DISCUSSION TITLE
I have a doubt

### DISCUSSION THREAD

**Discussion by smvybjh** [jflamy/owlcms4#1039] [General]
I have attached the image file, please look at it and tell if it is from Owlcms4 or some other software.

![WhatsApp Image 2024-07-10 at 1 53 20 PM](https://github.com/jflamy/owlcms4/assets/175200186/8e6af188-6828-4839-8de5-c49867200d01)

**Comment by smvybjh** [comment]
Where to get Referring device in budget

you attach the link.......

**Comment by jflamy** [comment]
See the documentation at https://owlcms.github.io/owlcms4/#/Refereeing

The cheapest way to have buttons is USB buttons https://owlcms.github.io/owlcms4/#/Refereeing?id=example-of-usb-keypads

Next you can build your own devices.  There are two ways of doing this
- many people have used joystick emulation https://owlcms.github.io/owlcms4/#/Refereeing?id=example-of-joystick-buttons together with the joy2key software.  This is cheap, but the joy2key software or similar is painful to use.
- for the same price, instead of getting a joystick emulation electronic board, you can get an Arduino Nano clone from China.  You can use the plans in https://github.com/jflamy/owlcms-firmata/tree/main/diagrams/referee for the referee devices (omit the buzzers and LED lights for a very simple device) and in https://github.com/jflamy/owlcms-firmata/tree/main/diagrams/timekeeper.
The Arduino-based designs come with the free software to run them.  Several people in Spain, India, Russia, Canada have used these plans.
This is what I use for my club.

Finally, you can use Bluetooth buttons, for example Flic2.  This has been used in Sweden.

**Comment by jflamy** [comment]
The picture is not owlcms.  It is some other system, perhaps an adaptation of the Asian Weightlifting Federation's system.
